ipfscluster.go

```diff
@@ -74,6 +74,8 @@ type IPFSConnector interface {
 	ID() (api.IPFSID, error)
 	Pin(context.Context, *cid.Cid, bool) error
 	Unpin(context.Context, *cid.Cid) error
+	PinPath(pinPath string, p api.Pin) error
```

Comment:
hey, these functions should be in cluster.go as part of the `Cluster` object (main component). The IPFSConnector should just offer something `Resolve(ctx, path string) *cid`.

I think that this `Resolve()` function should have the `IsJustAKey()` logic that you did below. Eventually, if needed, it should use the `doPost()` to perform a request to the IPFS daemon (see how other methods do it), parse the response, extract and decode the Cid and return it.

In `cluster.go` you would have something very similar to:

```
func (c *Cluster) PinPath(path string, pin api.Pin) error {
   ci, err := c.ipfsconn.Resolve(path)
   pin.Cid = ci
   return c.Pin(pin)
}
```

Once you're there, we can discuss the next steps